src/KeyVault/KeyVault/SecurityDomain/Cmdlets/RestoreSecurityDomain.cs

```diff
@@ -1,49 +1,159 @@
+        protected const string ByName = "ByName";
+        protected const string ByInputObject = "ByInputObject";
+        // protected const string ByResourceId = "ByResourceID";
+
+        [Parameter(HelpMessage = "Name of the managed HSM.", Mandatory = true, ParameterSetName = ByName)]
+        [Parameter(HelpMessage = "Name of the managed HSM.", Mandatory = true, ParameterSetName = ByRestoredBlob)]
+        [Parameter(HelpMessage = "Name of the managed HSM.", Mandatory = true, ParameterSetName = GenerateExchangeKey)]
+        [Alias("HsmName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(HelpMessage = "Object representing a managed HSM.", Mandatory = true, ParameterSetName = ByInputObject, ValueFromPipeline = true)]
+        [ValidateNotNull]
+        public PSKeyVaultIdentityItem InputObject { get; set; }
+
+        [Parameter(HelpMessage = "Information about the keys that are used to decrypt the security domain data. See examples for how it is constructed.", Mandatory = true, ParameterSetName = ByName)]
+        [Parameter(HelpMessage = "Information about the keys that are used to decrypt the security domain data. See examples for how it is constructed.", Mandatory = true, ParameterSetName = ByInputObject)]
+        [Parameter(HelpMessage = "Information about the keys that are used to decrypt the security domain data. See examples for how it is constructed.", Mandatory = true, ParameterSetName = DoRestoreBlob)]
         [ValidateNotNullOrEmpty]
         public KeyPath[] Keys { get; set; }
 
-        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true)]
+        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true, ParameterSetName = ByName)]
+        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true, ParameterSetName = ByInputObject)]
+        [Parameter(HelpMessage = "Specify the path to the prepared security domain data.", Mandatory = true, ParameterSetName = ByRestoredBlob)]
+        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true, ParameterSetName = DoRestoreBlob)]
         [Alias("Path")]
         [ValidateNotNullOrEmpty]
         public string SecurityDomainPath { get; set; }
 
+        [Parameter(HelpMessage = "Local file path to store the security domain encrypted with the exchange key.", Mandatory = true, ParameterSetName = DoRestoreBlob)]
+        [ValidateNotNullOrEmpty]
+        public string SecurityDomainRestoredBlob { get; set; }
+
+        [Parameter(HelpMessage = "Information about the exchange key used to encrypt the security domain data. Constructed by Initialize-AzKeyVaultSecurityDomainRecovery.", Mandatory = true, ParameterSetName=DoRestoreBlob)]
+        [ValidateNotNullOrEmpty]
+        public string ExchangeKey { get; set; }
```

Comment:
```suggestion
        [Parameter(HelpMessage = "Information about the exchange key used to encrypt the security domain data. Constructed by Initialize-AzKeyVaultSecurityDomainRecovery.", Mandatory = true, ParameterSetName=DoRestoreBlob)]
        [ValidateNotNullOrEmpty]
        public string ExchangeKeyInputPath { get; set; }
```

Reply:
Better not change, align with existed parameter Keys. And ExchangeKeyOutputFile is clear enough to identify the output and input ones. But this comment notified me with the incorrect help message, I will update the help message soon

Reply:
I think customer may wonder if we can pass the value of  ExchangeKey directly in this parameter. ExchangeKey means the path of ExchangeKey. It's reasonable to add Path in its naming to avoid confusion.

To be pair with ExchangeKeyOutputPath, that's why I'm proposing ExchangeKeyInputPath. ExchangeKey is different with Keys as the type of Keys is KeyPath but ExchangeKey is a string. 

Another proposal is let's combine ExchangeKeyInputPath and ExchangeKeyOutputPath as one parameter. If DownloadExchangeKey is present, we regard it as output path. Otherwise, it's input.

Reply:
I will do the combination